includes/db_specific/PostgreSql/QPgQuery.class.php

```diff
@@ -0,0 +1,224 @@
+		 * @param string       $strValue
+		 *
+		 * @throws Exception
+		 * @throws QCallerException
+		 * @throws QInvalidCastException
+		 */
+		public function __construct(QQColumnNode $objQueryNode, $strValue) {
+			parent::__construct($objQueryNode);
+			
+			if ($strValue instanceof QQNamedValue) {
+				$this->mixOperand = $strValue;
+			} else {
+				try {
+					$this->mixOperand = QType::Cast($strValue, QType::String);
+				} catch (QCallerException $objExc) {
+					$objExc->IncrementOffset();
+					$objExc->IncrementOffset();
+					throw $objExc;
+				}
+			}
+		}
+		
+		/**
+		 * @param QQueryBuilder $objBuilder
+		 */
+		public function UpdateQueryBuilder(QQueryBuilder $objBuilder) {
+			$mixOperand = $this->mixOperand;
+			if ($mixOperand instanceof QQNamedValue) {
+				/** @var QQNamedValue $mixOperand */
+				$objBuilder->AddWhereItem($this->objQueryNode->GetColumnAlias($objBuilder) . ' ILIKE ' . $mixOperand->Parameter());
+			} else {
+				$objBuilder->AddWhereItem($this->objQueryNode->GetColumnAlias($objBuilder) . ' ILIKE ' . $objBuilder->Database->SqlVariable($mixOperand));
+			}
+		}
+	}
+	
+	/**
+	 * Class QPgQConditionJsonbContainsValue: For performing '@>' JSON contains search in JSONB documents
+	 *
+	 */
+	class QPgQConditionJsonbContainsValue extends QQConditionComparison {
+		/**
+		 * @param QQColumnNode $objQueryNode
+		 * @param string       $strValue
+		 *
+		 * @throws Exception
+		 * @throws QCallerException
+		 * @throws QInvalidCastException
+		 */
+		public function __construct(QQColumnNode $objQueryNode, $strValue) {
+			parent::__construct($objQueryNode);
+			
+			if ($strValue instanceof QQNamedValue) {
+				$this->mixOperand = $strValue;
+			} else {
+				try {
+					$this->mixOperand = QType::Cast($strValue, QType::String);
+				} catch (QCallerException $objExc) {
+					$objExc->IncrementOffset();
+					$objExc->IncrementOffset();
+					throw $objExc;
+				}
+			}
+		}
+		
+		/**
+		 * @param QQueryBuilder $objBuilder
+		 */
+		public function UpdateQueryBuilder(QQueryBuilder $objBuilder) {
+			$mixOperand = $this->mixOperand;
+			if ($mixOperand instanceof QQNamedValue) {
+				/** @var QQNamedValue $mixOperand */
+				$objBuilder->AddWhereItem($this->objQueryNode->GetColumnAlias($objBuilder) . ' @> ' . $mixOperand->Parameter());
+			} else {
+				$objBuilder->AddWhereItem($this->objQueryNode->GetColumnAlias($objBuilder) . ' @> ' . $objBuilder->Database->SqlVariable($mixOperand));
+			}
+		}
+	}
+	
+	/**
+	 * Class QPgQConditionJsonbEqual: For comparing two JSONB documents for euqality
+	 * (if they are semantically equal, PostgreSQL considers them equal)
+	 */
+	class QPgQConditionJsonbEqual extends QQConditionComparison {
+		/**
+		 * @param QQColumnNode $objQueryNode
+		 * @param string       $strValue
+		 *
+		 * @throws Exception
+		 * @throws QCallerException
+		 * @throws QInvalidCastException
+		 */
+		public function __construct(QQColumnNode $objQueryNode, $strValue) {
```

Comment:
I am working on some changes, but just wanted to comment on this. This appears unnecessary here. Just use the standard QQ::Equals for this. Or am I missing something? Also, you don't need something special for jsonb vs json. The Postgres documentation says the two columns will produce the same results for queries, its just a matter of the internal storage format. JsonB allows postgres to store the json already parsed, whereas json type will force postgres to parse the json every time you do an operation. The results of the operation are identical, just not the performance.